include/modem/lte_lc.h

```diff
@@ -1099,20 +1151,25 @@ int lte_lc_lte_mode_get(enum lte_lc_lte_mode *mode);
  *	   The result of the measurement is reported back as an event of the type
  *	   LTE_LC_EVT_NEIGHBOR_CELL_MEAS, meaning that an event handler must be
  *	   registered to receive the information.
- *	   Depending on the network conditions and LTE connection state, it may
- *	   take a while before the measurement result is ready and reported back.
+ *	   Depending on the network conditions, LTE connection state and requested
+ *         search type, it may take a while before the measurement result is
+ *         ready and reported back.
  *	   After the event is received, the neighbor cell measurements
  *	   are automatically stopped.
  *
  * @note For modem firmware versions < v1.3.1, LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT
  *	 is the only accepted type. Other types will result in an error being returned.
+ *	 For modem firmware versions >= v1.3.4, also GCI search types and gci_count
+ *       are accepted.
  *
- * @param type Search type, see @c enum lte_lc_neighbor_search_type for more information.
+ * @param params Pointer to search type parameters or NULL to initiate a measurement
+ *               with the default parameters.
+ *               See @c struct lte_lc_ncellmeas_params for more information.
  *
  * @retval 0 if neighbor cell measurement was successfully initiated.
  * @retval -EFAULT if AT command failed.
  */
-int lte_lc_neighbor_cell_measurement(enum lte_lc_neighbor_search_type type);
+int lte_lc_neighbor_cell_measurement(struct lte_lc_ncellmeas_params *params);
```

Comment:
Refactoring the existing API looks a bit dangerous.

Can you just change the name of this new API. Otherwise API call that use `lte_lc_neighbor_cell_measurement(LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT)` would magically just work, because 0 is converted to NULL. But then any other value would crash.
Any enum value, however, would lead to compilation warning, but those are easy to miss.

I would instead propose that old API would stay the same, and this new would have a different name for example:
```c
int lte_lc_neighbor_cell_meas(struct lte_lc_ncellmeas_params *params);

/** Backward compatible wrapper for lte_lc_neighbor_cell_meas() */
inline int lte_lc_neighbor_cell_measurement(enum lte_lc_neighbor_search_type type)
{
    struct lte_lc_ncellmeas_params params = {
        .search_type = type
    };
    return lte_lc_neighbor_cell_meas(&params);
}
```

Reply:
A very good point! 

Yes, it compiles without warnings with enum value LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT (=0), but on the other hand; it also works as should with current NULL (=default).
And yes, if using any other enum value, it leads to compiler warnings and a runtime crash where a rootcause might not be seen easily, i.e. is not like assert() with row number.
Anyways, it is topic of another discussions, but IMO, compiler clean builds should be one of the commit entry criteria or at least, cause a CI failure at the 1st possible quality gate.

>I would instead propose that old API would stay the same
Yes, that is one possible option, others might be:
- revert how it was originally in this PR -> not a pointer, but a struct. i.e. no "NULL" defaulting feature.
- leave it as it is now.
- Other?

Any comments from code owners? @jtguggedal @simensrostad @tokangas?

Reply:
Good catch, Seppo. If only we had a way in C to check run-time if a variable is a valid pointer..

Not sure I'd want to change the function name. I see three alternatives:
* Not allowing NULL default or alternatively
* Let the function take two parameters and not use the struct. It's not optimal, but it will at least prevent unintentional use of enums.
* Option proposed by @jhirsi offline is to change the enum values. Then we could also assert or event do run-time check in the function that enum values are not used as input pointer.

Reply:
I also second keeping the function name as it is. I think adding an assert would be good, though. Even if values 1-5 would cause a crash, the reason for the crash is much more explicit with an assert. Changing the enum values is also fine by me, I would prefer asserting also when `LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT` is given. That would cause all users of the API to notice that the API has changed and update their implementation.

Reply:
OK. If the intention is to keep the name as it is. Then allow the `DEFAULT` and possibly add asserts to 1-5 values.

Reply:
IMO changing the enum values is the cleanest approach that would solve backward compatibility and prevent improper use of the function. I.E passing in an enumerator value to a function that accepts a pointer to a structure. That's is a bit weird and could confuse users.

Reply:
Thank you for the comments!
I have now pushed the change that changes enum values and additionally asserts for enum values.

Reply:
An alternative, if the `struct` is preferred for flexibility, would be to disallow `NULL` and instead provide a macro to initialize a default `struct` on the fly. Users would be forced to update, since `NULL` would return an error an 0 would emit an error during compilation.
```
void foobar(struct foo *);
#define DEFAULT &(struct foo){.abc= 0} 
foobar(DEFAULT)
```